src/instruction/instructions/store.ts

```diff
@@ -0,0 +1,302 @@
+import { Halfword, Word } from 'types/binary'
+import {
+  checkOptionCount,
+  create,
+  createImmediateBits,
+  createLowRegisterBits,
+  createRegisterBits,
+  getBits,
+  isImmediate,
+  isOptionCountValid, registerStringHasBrackets, removeBracketsFromRegisterString,
+  setBits
+} from 'instruction/opcode'
+import { Registers } from 'board/registers'
+import { IMemory } from 'board/memory/interfaces'
+
+import { ILabelOffsets } from '../interfaces'
+import { BaseInstruction } from './baseInstruction'
+
+/**
+ * Represents a 'STORE' instruction - STR (immediate offset) - word
+ */
+export class StoreInstructionImmediateOffset extends BaseInstruction {
+  public name: string = 'STR'
+  public pattern: string = '01100XXXXXXXXXXX'
+  private rnPattern: string = '0110000000XXX000'
+  private rtPattern: string = '0110000000000XXX'
+  private immPattern: string = '01100XXXXX000000'
+  private expectedOptionCount: number = 3
+
+  public canEncodeInstruction(commandName: string, options: string[]): boolean {
+    if (super.canEncodeInstruction(commandName, options)) {
+      if (
+          isOptionCountValid(options, this.expectedOptionCount) &&
+          isImmediate(options[2]) &&
+          registerStringHasBrackets(options[1],options[2])
+      ) {
+        return true
+      }
+    }
+    return false
+  }
+
+  public encodeInstruction(options: string[], labels: ILabelOffsets): Halfword {
+    checkOptionCount(options,3)
+    let opcode: Halfword = create(this.pattern)
+    opcode = setBits(opcode, this.rtPattern, createLowRegisterBits(removeBracketsFromRegisterString(options[0])))
```

Comment:
with this i can run STR [R2], [R3, #8] which is not valid.

Reply:
also for other classes do not remove brackets on first option

Reply:
But do we need to check this here? I think this is something which should be checked on the function canencodeinstruction. There we check if the brackets are set correct. This is only the function to remove them.

Reply:
in the canEncode you only check if option index 1 and 2 has correct set brackets (yeah that is intended to be there)

but on first param there must never be any brackets otherwise the command is invalid. I would check it in this method encodeInstruction by not removing and then no register will be found if a bracket is in place.